### ISSUE TITLE
CSS Scrollbars

### ISSUE THREAD

**Issue by rachelandrew** [mdn/sprints#522]
As a web developer I want to understand the CSS Scrollbars specification and browser support for the new properties.

## Acceptance criteria

Property pages for each property (I think is done, however I know the spec changed since I did them to combine properties).

Landing Page

Sort out the sidebar for related properties.

## Tasks

1. - [x] Create or update pages for each property
2. - [x] Create or update compat data for each property
3. - [x] Create a landing page
4. - [ ] Sort out sidebar

**Comment by rachelandrew** [comment]
Spec has changed as has the Nightly impl. so I need to update the work I've already done on this.

**Comment by chrisdavidmills** [comment]
Cool, nice one Rachel.

I talked to Tantek a little while ago, and he has this to say:

"BTW I need to get the MDN pages for https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-face-color and https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-track-color combined into a new https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-colors page"

The implementation bug is https://bugzilla.mozilla.org/show_bug.cgi?id=1492012

**Comment by rachelandrew** [comment]
> "BTW I need to get the MDN pages for https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-face-color and https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-track-color combined into a new https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-colors page"

yup, that's the plan :)

**Comment by rachelandrew** [comment]
I have renamed scrollbar-track-color to scrollbar-color and I'd like to redirect https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-face-color to that page as well but I don't think I have permissions to delete/redirect a page (or I just don't know how to do it).

**Comment by chrisdavidmills** [comment]
Even if a PR has been merged, you still have to wait for a production push of the data before it'll actually show up on the site. These usually happen about once a week.

I can help with the page redirect. Just to check what you need:

You need https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-face-color redirected to https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color?

You'll also want to move https://developer.mozilla.org/en-US/docs/scrollbar-color to https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color, as currently you've got it sitting on the top level of the site.

**Comment by rachelandrew** [comment]
Compat data PR https://github.com/mdn/browser-compat-data/pull/3024

**Comment by rachelandrew** [comment]
> You need https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-face-color redirected to https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color?

yup

> You'll also want to move https://developer.mozilla.org/en-US/docs/scrollbar-color to https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color, as currently you've got it sitting on the top level of the site.

Grr. Thought I had sorted that. Trying again ...

**Comment by rachelandrew** [comment]
Found the scrollbar-width PR which wasn't merged as it was waiting for the other PR which is now merged: https://github.com/mdn/data/pull/278

**Comment by chrisdavidmills** [comment]
Cool. I've done the redirect. For future reference, you do it by going into edit mode on the page you want to redirect, pressing the top-right-most button on the WYSIWYG toolbar, typing in the title of the page you want to redirect to, and then saving.

Your second page move worked perfectly.

**Comment by rachelandrew** [comment]
scrollbar-color data https://github.com/mdn/data/pull/318

**Comment by rachelandrew** [comment]
I think that once all the data is merged then this is done, landing page is here: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scrollbars

The scrollbar-width page isn't showing up in the menu, I think though this is the one with the PR still open (above).

**Comment by rachelandrew** [comment]
Since I did this work, parts of it have been flagged as obsolete and not on a standard track, including `scrollbar-color` which is the new renamed property. I would roll back these changes but I am wondering why they have been made.

It is on a standard track, here is the FPWD: https://www.w3.org/TR/css-scrollbars-1/

**Comment by rachelandrew** [comment]
Updated https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color as it already has the compat warning, so a further red warning is not required, also removed the obsolete tag as that makes no sense. I don't know how to stop it being obsolete in the sidebar.

**Comment by rachelandrew** [comment]
PR for Kumascript to change spec status https://github.com/mdn/kumascript/pull/978

**Comment by chrisdavidmills** [comment]
It just needed a shift + Refresh to get rid of the obsolete tag in the sidebar. It generates that if the obsolete tag is present, so won't have it now you've removed it, but the page needed regenerating to seal the deal.

I guess since this is more in flux than we thought, we ought to leave it for now until it stabilizes?

**Comment by rachelandrew** [comment]
ah cool - I don't think it is in more flux that anything else at FPWD and as Firefox has implemented it makes sense to have the docs. If I can figure out why scrollbar-width isn't showing the data properly and fix it in the sidebar then this is pretty much done anyway as much as it needs to be.

If I hear anything else about it in WG meetings I can always fix it up :)

**Comment by chrisdavidmills** [comment]
It looks like `scrollbar-width` does not yet have an entry in https://github.com/mdn/data/blob/master/css/properties.json — I think that's the key to the problem.

I've given the `scrollbar-width` and `scrollbar-color` pages a tag of "CSS Scrollbars" — the group name they should both have in the data repo. Once they both have an entry in the data repo, with that group name, they should start to appear on each other's sidebars in the pages.

**Comment by rachelandrew** [comment]
I'm just found the PR, which seems to have conflicts. I'll see if I can sort it out today.

**Comment by chrisdavidmills** [comment]
OK, cheers Rachel.

**Comment by rachelandrew** [comment]
it got into a mess, so here is a new one: https://github.com/mdn/data/pull/322

**Comment by rachelandrew** [comment]
So scrollbar-width is in the data but still not showing up in the sidebar. I'm leaving this open until it shows up but I don't know what else to try.

**Comment by chrisdavidmills** [comment]
It just needed a Shift + Refresh to force regeneration of the pages. If you look at https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scrollbars and the property pages, you'll see that both now show.
